backend/src/gateway/index.ts

```diff
@@ -1,37 +1,71 @@
 import Messaging from "../models/Messages";
 import { Server } from "socket.io";
 import { Server as HTTPServer } from "http";
+import GroupChats from "../models/Groupchat";
 
 class Gateway {
-	socketIO: Server;
-	constructor(http: HTTPServer) {
-		this.socketIO = new Server(http);
-		console.log("Gateway initialized");
-	}
-
-	init() {
-		this.socketIO.on("connection", (socket) => {
-			console.log(`SOCKETIO: ${socket.id} user just connected!`);
-			Messaging.find().then((result) => {
-				socket.emit("output-messages", result);
-			});
-
-			socket.on("message", this.messageHandler.bind(this, socket));
-
-			socket.on("disconnect", () => {
-				console.log(`${socket.id} user disconnected`);
-			});
-		});
-	}
-
-	messageHandler(socket, msg) {
-		const message = new Messaging({ msg });
-		message.save().then((document) => {
-			console.log(document)
-			console.log("SAVING MESSAGE \"" + msg +"\" from user " + socket.id);
-			this.socketIO.emit("message", msg);
-		});
-	}
+  socketIO: Server;
+  constructor(http: HTTPServer) {
+    this.socketIO = new Server(http);
+    console.log("Gateway initialized");
+  }
+
+  init() {
+    this.socketIO.use((socket, next) => {
+      const username = socket.handshake.auth.username;
+      if (!username) {
+        return next(new Error("invalid username"));
+      }
+      (socket as any).username = username;
+      next();
+    });
+
+    this.socketIO.on("connection", (socket) => {
+      console.log(`SOCKETIO: ${socket.id} user just connected!`);
+
+      socket.on("send_message", this.onSendMessage.bind(this, socket));
+
+      Messaging.find().then((result) => {
```

Comment:
not totally sure what this is for, I guess it is to update the user for his new messages?
I guess that's fine for now but I recommend adding a TODO comment in the code just so that we know to change this to a better system later